analyzers/src/SonarAnalyzer.Common/RegularExpressions/RegexNode.cs

```diff
@@ -0,0 +1,94 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+#nullable enable
+
+using System.Text.RegularExpressions;
+
+namespace SonarAnalyzer.RegularExpressions;
+
+internal sealed class RegexNode
+{
+    public RegexNode(
+        SyntaxNodeValue<string?> pattern,
+        SyntaxNodeValue<RegexOptions?> options)
+    {
+        Pattern = pattern;
+        Options = options;
+    }
+
+    public SyntaxNodeValue<string?> Pattern { get; }
+    public SyntaxNodeValue<RegexOptions?> Options { get; }
```

Comment:
I don't think we need the `SyntaxNodeValue`. `PatternNode`, `Pattern`, `OptionsNode` and `Options` will do the trick just as fine

Reply:
I have a preference to combine them. The have a (very strong) relationship. But having 4 properties is also an option.

Reply:
It adds a level of indirection that isn't practically needed. Let's split it